src/main/java/com/google/sps/servlets/RouteImage.java

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.*;
+import com.google.sps.data.Images;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
+@SuppressWarnings("serial")
+public class RouteImage extends HttpServlet {
+  /** Store route's image. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+      String routeId = request.getParameter("name-route");
+      Key routeKey = KeyFactory.createKey("Route", Long.parseLong(routeId));
+      Entity routeEntity = datastore.get(routeKey);
+      String fileName = (String) routeEntity.getProperty("imageName");
+      if (fileName.equals("globe.jpg")) {
```

Comment:
Is it a specific image? What if the user will make their own image have the same name?
Also I think it should be a constant too

Reply:
It is default image i loaded to google cloud, all user's images will be saved under the name (user's id as a string) + ".png" so there shouldn't be any intersections.